receiver/jmxreceiver/README.md

```diff
@@ -47,7 +47,13 @@ The JMX Service URL the Metric Gatherer's JMX client should use.
 
 Corresponds to the `otel.jmx.service.url` property.
 
-_Required._
+One of `service_url` or `endpoint` is _required._.  Both cannot be specified at the same time.
+
+### endpoint
+The host and port used to construct the JMX Service URL the Metric Gatherer's JMX client should use.
+Value in the form of `"host:port"` will be used to create a `service_url` of `service:jmx:rmi:///jndi/rmi://<host>:<port>/jmxrmi`.
+
+One of `service_url` or `endpoint` is _required._.  Both cannot be specified at the same time.
```

Comment:
Why not using just one property that supports both formats?

Reply:
Would the logic be if the scheme is empty treat it as an endpoint and construct the service url and if there's a scheme already assume it's a service url?

Reply:
Something like that, if it starts with "service:" then is a service_url, otherwise an endpoint "host:port". Not sure if it works but trying to see if that works.

Reply:
I don't think the obligation of parsing and validating JMX service urls should be in the collector, which may have to be the case if `service_url` and `endpoint` are combined.  It would at minimum confirm service urls start with `"service:jmx"` in order to otherwise assume an `endpoint` is being specified.  That then increases the burden of endpoint validation.

The `endpoint` field provides compatibility with the receiver creator and is otherwise a separate ease of use feature.  Not sure what would be gained for either the user or us if they were combined.  Open to the alternative though.

Reply:
Downside with two properties is also users will get confused about strict requirements of only one being set, and most likely will get that wrong. I am ok with collector doing more work/validation and return clear errors to the user. So I would vote that we do the validation and use only one property so that the user don't get confused.

Reply:
endpoint is coerced to become a service url and only exists in the jmx receiver to be used by receiver creator and optionally by users.  The service url is passed as the value of the actual java property used by the jmx metric gatherer, where ["actual" validation occurs](https://docs.oracle.com/javase/7/docs/api/javax/management/remote/JMXServiceURL.html).

I will try combining them, but am not immediately sure stdlib provides [rfc 2609](https://tools.ietf.org/html/rfc2609) parsing abilities.

Reply:
@bogdandrutu @jrcamp updated to obsolete `service_url`